src/tiledb/cloud/bioimg/helpers.py

```diff
@@ -41,10 +41,18 @@ def is_folder(path: str) -> bool:
     return path.endswith("/")
 
 
-def validate_io_paths(source: Sequence[str], output: Sequence[str]) -> None:
+def validate_io_paths(
+    source: Sequence[str], output: Sequence[str], register: bool
+) -> None:
     if len(source) == 0 or len(output) == 0:
         raise ValueError("Source/Output list must not be empty.")
 
+    if register:
+        if any(o.startswith("tiledb://") for o in output):
+            raise ValueError(
+                "Invalid value of argument 'register'. \
+                             TileDB URIs require register argument to be set to False"
+            )
```

Comment:
@ktsitsi do you think we could design the ingestion so that `tiledb://` URIs and `register=True` condition can't occur?

Reply:
We offer here alternative paths to registration. When a user provides tiledb uris as outputs, then the registration is being performed during ingestion, as we pre-register the assets before writing to them. In this case we do not need the extra registration stage  inside the DAG. Even though these two could co-exist, in this scenario we would be adding extra registation nodes to the graph that would complete with `Dataset already registered`. The registration stage will not fail but will be useless so thus I am deactivating when tiledb uris are present and we inform the user.

Reply:
Indeed there will be eventually a change of design, where we will require only TileDB URIs and the register stage could be dropped. So you are correct here, but since this is a customer facing API we need to keep it as backwards compatible as possible for the time being and don't affect customer usage, before we communicate changes.